cheddar/cheddar-events/src/main/java/com/clicktravel/cheddar/event/EventMessageHandler.java

```diff
@@ -40,20 +42,30 @@ public void handle(final TypedMessage typedMessage) throws MessageHandlingExcept
         try {
             final String messageType = typedMessage.getType();
             logger.debug("Handling: " + messageType);
-            final EventHandler<E> eventHandler = eventHandlers.get(messageType);
-            final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
-            if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
-                throw new IllegalStateException("Event of type " + event.getClass() + " is not compatible with "
-                        + eventHandler.getEventClass() + " in event handler");
+            for (final EventHandler<E> eventHandler : eventHandlers.get(messageType)) {
+                try {
+                    final E event = AbstractEvent.newEvent(eventHandler.getEventClass(), typedMessage.getPayload());
+                    if (!eventHandler.getEventClass().isAssignableFrom(event.getClass())) {
+                        throw new IllegalStateException("Event of type " + event.getClass()
+                                + " is not compatible with " + eventHandler.getEventClass() + " in event handler");
+                    }
+                    eventHandler.handle(event);
+                } catch (final Exception e) {
+                    logger.error("Error handling message: " + typedMessage, e);
```

Comment:
now we are just catching this will we ever hit the MessageHandlingException below?

and does this impact the retry logic?